### PR TITLE
Db exception detail

### PR DESCRIPTION
**Pull request by psa**
Make it easier to debug write attempt failures by logging the error from SQLite.

Check List

- [x] I have read CONTRIBUTING.md and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both setup.cfg (and conda-environment.yml if present).
- [x] Tests are included (or explain why tests are not needed).
    - Simple text change in error logging
- [x] Changelog entry included if this is a change that can affect users
    - Trivial usability improvement
- [x] Docs not needed
- [x] If this is a bug fix, PR should be raised against the relevant ?.?.x branch.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
(closing and re-opening to sort the tests out)

**Review by oliver-sanders — APPROVED** [review]
LGTM, tested as working.

@psa, did you get to the bottom of the write error?

File: cylc/flow/rundb.py [hunk 1]
```diff
@@ -493,8 +493,8 @@ def execute_queued_items(self):
                 raise
             self.n_tries += 1
             LOG.warning(
-                "%(file)s: write attempt (%(attempt)d) did not complete\n" % {
-                    "file": self.db_file_name, "attempt": self.n_tries})
+                "%(file)s: write attempt (%(attempt)d) did not complete: %(error)s\n" % {
+                    "file": self.db_file_name, "attempt": self.n_tries, "error": str(e)})
             if self.conn is not None:
```
Comment (oliver-sanders) [inline cylc/flow/rundb.py:498]: These lines just needed a little reformatting to fit within the project line length

```suggestion
            LOG.warning(
                "%(file)s: write attempt (%(attempt)d)"
                " did not complete: %(error)s\n" % {
                    "file": self.db_file_name,
                    "attempt": self.n_tries,
                    "error": str(e)
                }
            )
            if self.conn is not None:
```

File: cylc/flow/rundb.py [hunk 2]
```diff
@@ -493,8 +493,8 @@ def execute_queued_items(self):
                 raise
             self.n_tries += 1
             LOG.warning(
-                "%(file)s: write attempt (%(attempt)d) did not complete\n" % {
-                    "file": self.db_file_name, "attempt": self.n_tries})
+                "%(file)s: write attempt (%(attempt)d) did not complete: %(error)s\n" % {
+                    "file": self.db_file_name, "attempt": self.n_tries, "error": str(e)})
```
Comment (oliver-sanders) [inline cylc/flow/rundb.py:497]: These lines just needed a little reformatting to fit within the project line length:

```suggestion
                "%(file)s: write attempt (%(attempt)d)"
                " did not complete: %(error)s\n" % {
                    "file": self.db_file_name,
                    "attempt": self.n_tries,
                    "error": str(e)
                }
            )
```

**Comment by psa** [comment]
> LGTM, tested as working.
> 
> @psa, did you get to the bottom of the write error?

We're still working on it, it only shows up every month or so, so it's been a bit of a pain to debug.

**Pull request merged**